contracts/tgrade-valset/src/contract.rs

```diff
@@ -667,8 +670,31 @@ fn end_block<Q: CustomQuery>(deps: DepsMut<Q>, env: Env) -> Result<Response, Con
     epoch.current_epoch = cur_epoch;
     EPOCH.save(deps.storage, &epoch)?;
 
+    if cfg.verify_validators {
+        let votes = deps
+            .querier
+            .query::<ValidatorVoteResponse>(&QueryRequest::Custom(TgradeQuery::ValidatorVotes {}))?
+            .votes;
+
+        if let Some(pending) = PENDING_VALIDATORS.may_load(deps.storage)? {
+            let expiration = JailingPeriod::from_duration(
+                JailingDuration::Duration(cfg.offline_jail_duration),
+                &env.block,
+            );
+
+            for val in pending {
+                let vote = votes
+                    .iter()
+                    .find(|vote| vote.address.as_slice() == val.1.to_address());
```

Comment:
Building a hash map of validator votes per validator address can be an option here.

It only makes sense if there can be or there usually would be many (more than our around, let's say 5) pending validators at the same time, though.